ui/storybook/README.md

```diff
@@ -0,0 +1,122 @@
+
+## Getting started
+
+### Running storybook
+
+- Run storybook in development mode: `npm run storybook`
+
+### Building storybook
+
+- Build only: `npm run storybook:build`
+- Build & serve static assets from build: `npm run storybook:serve`
+
+## Creating stories
+
+### Where should stories live?
+
+Stories **SHOULD** live in the first entry in the list below that matches their purpose:
+
+- Stories that are tightly coupled with a component should be colocated with the component code in the associated package (e.g. the stories for the `LineChart` component live at `ui/components/src/LineChart/LineChart.stories.tsx` as a sibling of `LineChart.tsx`).
+- Package-specific stories should live in the `src/stories` directory for that package.
+- Stories that provide project-wide documentation or involve components from multiple packages should live in the storybook package (`ui/storybook/stories`).
+
+### What kind of story format should I use?
+
+#### Components
+
+Stories for components should use [Component Story Format (CSF)](https://storybook.js.org/docs/react/api/csf) and follow the `PascalCaseName.stories.tsx` naming scheme.
+
+See [Writing stories in TypeScript](https://storybook.js.org/blog/writing-stories-in-typescript/) for some additional guidance on typing for stories.
+
+#### Text-only documentation
+
+Stories that are primarily text-based documentation (e.g. the `About` pages for each package) should be written using [MDX](https://storybook.js.org/docs/react/writing-docs/mdx) and follow the `PascalCase.stories.mdx` naming scheme.
+
+These may include importing a few small components like the `LinkTo` component from the [Link addon](https://storybook.js.org/addons/@storybook/addon-links/) to link out to other stories.
+
+### Importing package code
+
+#### TLDR;
+
+Stories should import package code by referencing the package, not by making a relative reference. Under the hood, this will reference source code (not compiled code), so we can benefit from some Storybook features like autogenerating documentation from types.
+
+Import like this in a story:
+
+```ts
+// Import from the package. Storybook's webpack config will alias this to the
+// associated source code.
+import { LineChart } from '@perses-dev/components';
+```
+
+Not like this:
+
+```ts
+// Importing directly from the source file will make it easier to forget to
+// export components for consumption from the main export for the package.
+import { LineChart } from '.';
+```
+
+#### Long version
+
+Stories should import from source code (NOT compiled code) because this enables us to automatically generate documentation tables for component props from types. Stories should import from the top level export for that package to help ensure the component was properly exported for consumption from the package when published. The same versions of code need to be used across the storybook to avoid issues when using `context` and singletons.
```

Comment:
It's a bummer we need to do this, but I think having the docs outweighs the downside by a lot. So I agree.